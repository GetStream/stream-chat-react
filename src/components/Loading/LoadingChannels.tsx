import React from 'react';

const LoadingItems = () => (
  <div className='str-chat__loading-channels-item str-chat__channel-preview-loading'>
    <div className='str-chat__loading-channels-avatar' />
    <div className='str-chat__loading-channels-meta str-chat__channel-preview-end-loading'>
      <div className='str-chat__loading-channels-username' />
      <div className='str-chat__loading-channels-status' />
    </div>
  </div>
);

const UnMemoizedLoadingChannels = () => (
  <div className='str-chat__loading-channels'>
    <LoadingItems />
    <LoadingItems />
    <LoadingItems />
  </div>
);

/**
 * Loading indicator for the ChannelList
 */
export const LoadingChannels = React.memo(UnMemoizedLoadingChannels);
