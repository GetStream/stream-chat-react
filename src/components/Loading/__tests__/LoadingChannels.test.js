import React from 'react';

import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { LoadingChannels } from '../LoadingChannels';

afterEach(cleanup);

describe('LoadingChannels', () => {
  it('should render component with default props', () => {
    const { container } = render(<LoadingChannels />);
    expect(container).toMatchSnapshot();
  });
});
