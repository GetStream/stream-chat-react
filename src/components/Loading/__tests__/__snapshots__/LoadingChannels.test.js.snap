// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingChannels should render component with default props 1`] = `
<div
  className="str-chat__loading-channels"
>
  <div
    className="str-chat__loading-channels-item str-chat__channel-preview-loading"
  >
    <div
      className="str-chat__loading-channels-avatar"
    />
    <div
      className="str-chat__loading-channels-meta str-chat__channel-preview-end-loading"
    >
      <div
        className="str-chat__loading-channels-username"
      />
      <div
        className="str-chat__loading-channels-status"
      />
    </div>
  </div>
  <div
    className="str-chat__loading-channels-item str-chat__channel-preview-loading"
  >
    <div
      className="str-chat__loading-channels-avatar"
    />
    <div
      className="str-chat__loading-channels-meta str-chat__channel-preview-end-loading"
    >
      <div
        className="str-chat__loading-channels-username"
      />
      <div
        className="str-chat__loading-channels-status"
      />
    </div>
  </div>
  <div
    className="str-chat__loading-channels-item str-chat__channel-preview-loading"
  >
    <div
      className="str-chat__loading-channels-avatar"
    />
    <div
      className="str-chat__loading-channels-meta str-chat__channel-preview-end-loading"
    >
      <div
        className="str-chat__loading-channels-username"
      />
      <div
        className="str-chat__loading-channels-status"
      />
    </div>
  </div>
</div>
`;
